crypto/provider_core.c

```diff
@@ -572,67 +572,77 @@ static int provider_forall_loaded(struct provider_store_st *store,
     return ret;
 }
 
+/*
+ * This function only does something once when store->use_fallbacks == 1,
+ * and then sets store->use_fallbacks = 0, so the second call and so on is
+ * effectively a no-op.
+ */
+static void provider_activate_fallbacks(struct provider_store_st *store)
+{
+    if (store->use_fallbacks) {
```

Comment:
Other places where we read `store->use_fallbacks` we are protected by the store lock. However if this function is called from `ossl_provider_available` then that isn't the case.

Reply:
Fixed

Reply:
Errr....is this fixed???

Reply:
Yes.  All calls to this function are guarded with a lock at this point.